src/email.c

```diff
@@ -0,0 +1,184 @@
+#define _GNU_SOURCE 1
+
+#include "config.h"
+
+#include <errno.h>
+#include <locale.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <gtk/gtk.h>
+
+#include <gio/gio.h>
+#include <gio/gdesktopappinfo.h>
+#include <gio/gunixfdlist.h>
+
+#include "xdg-desktop-portal-dbus.h"
+#include "shell-dbus.h"
+
+#include "email.h"
+#include "request.h"
+#include "utils.h"
+
+static gboolean
+compose_mail_evolution (const char  *address,
+                        const char  *subject,
+                        const char  *body,
+                        const char **attachments,
+                        GError     **error)
+{
+  g_autofree char *enc_subject = NULL;
+  g_autofree char *enc_body = NULL;
+  g_autoptr(GString) url = NULL;
+  g_autoptr(GSubprocess) subprocess = NULL;
+  const char *argv[4];
+  int i;
+  g_autofree char *command = NULL;
+
+  enc_subject = g_uri_escape_string (subject ? subject : "", NULL, FALSE);
+  enc_body = g_uri_escape_string (body ? body : "", NULL, FALSE);
+
+  url = g_string_new ("mailto:");
+
+  g_string_append_printf (url, "\"%s\"", address ? address : "");
+  g_string_append_printf (url, "?subject=%s", enc_subject);
+  g_string_append_printf (url, "&body=%s", enc_body);
+
+  for (i = 0; attachments[i]; i++)
+    {
+      g_autofree char *path = g_uri_escape_string (attachments[i], NULL, FALSE);
+      g_string_append_printf (url, "&attach=%s", path);
+    }
+
+  argv[0] = "evolution";
```

Comment:
I think a better idea would be to use something like _xdg-email_ instead of using hardcoded email client(s).